docs/src/modules/spring/pages/event-sourced-entities.adoc

```diff
@@ -0,0 +1,160 @@
+
+
+NOTE: The entity key `cartId` must match a path parameter (i.e. `cartId`) and such value needs to be unique per entity. On the other hand, the entity type `shopping-cart` is common for all instances of this entity but should be unique across the different entity types.
+
+
+
+=== Updating state
+
+Having created the basis of our entity, we will now define how each command is handled. In the example below, we define a new endpoint that will add a new line item to a given shopping cart. It returns an `Effect` to emit an event and then sends a reply once the event is stored successfully. The state is updated by the event handler.
+
+IMPORTANT: The **only** way for a command handler to modify the Entity's state is by emitting an event. Any modifications made directly to the state (or instance variables) from the command handler are not persisted. When the Entity is passivated and reloaded, those modifications will not be present.
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/ShoppingCartService.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartService.java[tag=addItem]
+----
+<1> The validation ensures the quantity of items added is greater than zero and it fails for calls with illegal values by returning an `Effect` with `effects().error`.
+<2> From the current incoming `LineItem` we create a new `ItemAdded` event representing the change of the cart.
+<3> We store the event by returning an `Effect` with `effects().emitEvent`.
+<4> The acknowledgment that the command was successfully processed is only sent if the event was successfully stored, otherwise there will be an error reply.
+<5> Event handler needs to be marked with `@EventHandler` and receive a single parameter type matching the event type produced (i.e. `ItemAdded`).
+<6> Return the new state to be stored - the logic for state transition is defined inside the `ShoppingCart` domain model.
+
+
+NOTE: There needs to be one event handler declared per each type of event the ES entity emits (e.g. `itemAdded` receives a parameter of type `ItemAdded`, the same type emitted in `addItem` command handler).
+
+As mentioned above, the business logic that allows us to transition between states was placed on the domain model as seen below:
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=itemAdded]
+----
+<1> For an existing item, we will make sure to sum the existing quantity with the incoming one.
+<2> Returns an update list of items without the existing item.
+<3> Adds the update item to the shopping cart.
+<4> Returns a new instance of the shopping cart with the updated line items.
+
+- the `emptyState()` override is optional. If not using it, be careful to deal with a currentState() with a `null` value when receiving the first command
```

Comment:
Also when applying the first event. Here users are expected to initialize the state for the first time.